scripts/macos/agent_diagnostics.sh

```diff
@@ -0,0 +1,145 @@
+#!/bin/bash
+
+# agent_diagnostics.sh collects all artifacts required for troubleshooting,
+# including logs, configuration files, and application files. Those artifacts
+# are compressed into a zip file, along with an output file of relevant information
+# about the system for quick referrence and an inventory of the files included
+# in the zip file.
+
+if [[ "${UID}" != 0 ]]; then
+  (>&2 echo "Error:  $0 must be run as root")
+  exit 1
+fi
+
+# Some global variables
+JCPATH="/opt/jc"
+JCLOG="/var/log/"
+STAMP=$( date +"%Y%m%d%H%M%S" )
+ZIPFILE="./jc${STAMP}.zip"
+declare -a INVENTORY
+
+# Is jcagent installed? if not, exit.
+if [[ ! -d "${JCPATH}" ]]; then
+  echo "JCAGENT IS NOT INSTALLED ON THIS MACHINE."
+  exit 1
+fi
+
+# Is zip installed?
+if ! which zip 2> /dev/null; then
+  ZPATH="false"
+else
+  ZPATH=$(which zip)
+fi
+
+function indent() {
+	sed 's/^/		/g'
```

Comment:
I believe this should be ` sed 's/^/\t\t/g'` as you have in the Linux script.

Reply:
Due to a difference between bsd sed and gnu sed, I used a literal (ctrl-v + tab), originally. I changed this to be more readable with:
sed 's/^/\'$'\t/g'